client/src/components/categorical/category.js

```diff
@@ -33,17 +19,15 @@ class Category extends React.Component {
       isChecked: true,
       isExpanded: false
     };
-    this.countCategories = memoize((values, optsAsBools) =>
-      countCategories(values, optsAsBools)
-    );
   }
 
   componentDidUpdate() {
-    const { categoricalAsBooleansMap, metadataField, values } = this.props;
-    const categoryCount = this.countCategories(
-      values,
-      categoricalAsBooleansMap[metadataField]
-    );
+    const { categoricalSelectionState, metadataField } = this.props;
+    const cat = categoricalSelectionState[metadataField];
+    const categoryCount = {
+      total: cat.numOptions,
+      on: _.reduce(cat.optionSelected, (res, cond) => (cond ? res + 1 : res), 0)
```

Comment:
Hmm, feeling `on` might be a little bit obfuscated. Does it mean 'currentlySelected', or is it related to a handler for an event? Room for a better name or a comment here. Obvious what it's doing here after spending a minute reading the ternary, so maybe it's fine.

Reply:
will improve the clarity